platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -64,47 +68,157 @@
   private final int maximumShardsPerNode =
       NumberUtils.toInt(System.getProperty("solr.cloud.maxShardPerNode"), 2);
 
+  private static final String ALIAS_PROP = "alias";
+
   @Override
   public org.codice.solr.client.solrj.SolrClient newClient(String core) {
+    return newClient(core, null);
+  }
+
+  @Override
+  public org.codice.solr.client.solrj.SolrClient newClient(
+      String collection, Map<String, Object> properties) {
+
     String zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+    checkConfig(zookeeperHosts);
 
-    if (StringUtils.isBlank(zookeeperHosts)) {
-      LOGGER.warn(
-          "Cannot create SolrCloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
-    }
     LOGGER.debug(
-        "Solr({}): Creating a SolrCloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
+        "Solr({}): Creating a Solr Cloud client with configuration using Zookeeper hosts [{}]",
+        collection,
+        zookeeperHosts);
+
+    String alias = properties == null ? null : (String) properties.getOrDefault(ALIAS_PROP, null);
+
     return new SolrClientAdapter(
-        core,
+        collection,
         () ->
             AccessController.doPrivileged(
                 (PrivilegedAction<SolrClient>)
                     () -> {
-                      return createSolrCloudClient(zookeeperHosts, core);
+                      return createSolrCloudClient(zookeeperHosts, collection, alias);
                     }));
   }
 
   @VisibleForTesting
-  SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
+  void addCollectionToAlias(String alias, String collection, CloudSolrClient client) {
+    if (StringUtils.isBlank(alias) || StringUtils.isBlank(collection)) {
+      return;
```

Comment:
❓ Just confirming, is silently returning (instead of throwing an exception) what we want here if the alias or collection is null?

Reply:
going to remove the null check here. https://github.com/lamhuy/ddf/blob/preSplit/platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java#L212
making this error benign. It just mean noop.